test/Microsoft.AspNetCore.Hosting.FunctionalTests/ShutdownTests.cs

```diff
@@ -0,0 +1,88 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Diagnostics;
+using System.IO;
+using System.Net.Http;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Server.IntegrationTesting;
+using Microsoft.AspNetCore.Server.IntegrationTesting.xunit;
+using Microsoft.AspNetCore.Testing.xunit;
+using Microsoft.Extensions.Logging;
+using Xunit;
+using Xunit.Sdk;
+
+namespace Microsoft.AspNetCore.Hosting.FunctionalTests
+{
+    public class ShutdownTests
+    {
+        [ConditionalFact]
+        [OSSkipCondition(OperatingSystems.Windows)]
+        [OSSkipCondition(OperatingSystems.MacOSX)]
+        public void ShutdownTest()
+        {
+            var logger = new LoggerFactory()
+                .AddConsole()
+                .CreateLogger(nameof(ShutdownTest));
+
+            string applicationPath = Path.Combine(TestProjectHelpers.GetProjectRoot(), "..",
+                "Microsoft.AspNetCore.Hosting.TestSites");
+
+            var deploymentParameters = new DeploymentParameters(
+                applicationPath,
+                ServerType.Kestrel,
+                RuntimeFlavor.CoreClr,
+                RuntimeArchitecture.x64)
+            {
+                EnvironmentName = "Shutdown",
+                TargetFramework = "netcoreapp1.1",
+                ApplicationType = ApplicationType.Portable,
+                PublishApplicationBeforeDeployment = true
+            };
+
+            using (var deployer = new SelfHostDeployer(deploymentParameters, logger))
+            {
+                deployer.Deploy();
+
+                // Wait for application to start
+                System.Threading.Thread.Sleep(1000);
+
+                string output = string.Empty;
+                deployer.HostProcess.OutputDataReceived += (sender, args) => output += args.Data + '\n';
+
+                SendSIGINT(deployer.HostProcess.Id);
+
+                deployer.HostProcess.WaitForExit();
```

Comment:
Is this going to make the test hang if something goes wrong? Any way to timeout here?